extension/parquet/parquet_crypto.cpp

```diff
@@ -195,19 +191,31 @@ class DecryptionTransport : public TTransport {
 	}
 
 	uint32_t Finalize() {
+
 		if (read_buffer_offset != read_buffer_size) {
-			throw InternalException("DecryptionTransport::Finalize was called with bytes remaining in read buffer");
+			throw InternalException("DecryptionTransport::Finalize was called with bytes remaining in read buffer: \n"
+			                        "read buffer offset: %d, read buffer size: %d",
+			                        read_buffer_offset, read_buffer_size);
 		}
 
 		data_t computed_tag[ParquetCrypto::TAG_BYTES];
-		if (aes.Finalize(read_buffer, AESGCMState::BLOCK_SIZE, computed_tag, ParquetCrypto::TAG_BYTES) != 0) {
-			throw InternalException("DecryptionTransport::Finalize was called with bytes remaining in AES context");
+
+		if (aes->IsOpenSSL()) {
```

Comment:
Is there no way of unifying this in the callbacks? That would be cleaner as we can keep the logic in a single location